Problem during upload - File descriptor??

Currently unable to upload data sets to frisian-stories server.

11:08:52,219 INFO  ~ Accepting file 8ABCAAD9C62D57BBC0191ABFE79A7E90__hints.txt for DataSet FriesMuseum
11:08:52,439 ERROR ~ 

@680kc8h12
Internal Server Error (500) for request POST /frisianstories/dataset/sip-creator/submit/FriesMuseum/8ABCAAD9C62D57BBC0191ABFE79A7E90__hints.txt?accessKey=27281a59745247fc8a870296a87542

Oops: IOException
An unexpected error occured caused by exception IOException: Bad file descriptor

play.exceptions.UnexpectedException: Unexpected Error
        at play.libs.IO.readContent(IO.java:185)
        at controllers.SipCreatorEndPoint$.receiveHints(SipCreatorEndPoint.scala:183)
        at controllers.SipCreatorEndPoint$.acceptFile(SipCreatorEndPoint.scala:145)
        at play.mvc.ActionInvoker.invokeWithContinuation(ActionInvoker.java:548)
        at play.mvc.ActionInvoker.invoke(ActionInvoker.java:502)
        at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:496)
        at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:473)
        at play.mvc.ActionInvoker.invoke(ActionInvoker.java:161)
        at Invocation.HTTP Request(Play!)
Caused by: java.io.IOException: Bad file descriptor
        at java.io.FileInputStream.readBytes(Native Method)
        at java.io.FileInputStream.read(FileInputStream.java:198)
        at play.libs.IO.readContent(IO.java:180)
        ... 8 more

I investigated this now for nearly 2 hours and ended up writing to the
Play ML. In short, it looks like the request.body can't be accessed
directly anymore, and I have a hard time figuring out why. The data is
there somewhere, albeit as a "body" request parameter, which is
unsuitable for parsing INTs and large data (source). I hope to get a
quick answer to that one.

On Sat, Oct 1, 2011 at 2:21 PM, Gerald de Jong
reply@reply.github.com
wrote:

> Currently unable to upload data sets to frisian-stories server.
> 
> 11:08:52,219 INFO  ~ Accepting file 8ABCAAD9C62D57BBC0191ABFE79A7E90__hints.txt for DataSet FriesMuseum
> 11:08:52,439 ERROR ~
> 
> @680kc8h12
> Internal Server Error (500) for request POST /frisianstories/dataset/sip-creator/submit/FriesMuseum/8ABCAAD9C62D57BBC0191ABFE79A7E90__hints.txt?accessKey=27281a59745247fc8a870296a87542
> 
> Oops: IOException
> An unexpected error occured caused by exception IOException: Bad file descriptor
> 
> play.exceptions.UnexpectedException: Unexpected Error
>        at play.libs.IO.readContent(IO.java:185)
>        at controllers.SipCreatorEndPoint$.receiveHints(SipCreatorEndPoint.scala:183)
>        at controllers.SipCreatorEndPoint$.acceptFile(SipCreatorEndPoint.scala:145)
>        at play.mvc.ActionInvoker.invokeWithContinuation(ActionInvoker.java:548)
>        at play.mvc.ActionInvoker.invoke(ActionInvoker.java:502)
>        at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:496)
>        at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:473)
>        at play.mvc.ActionInvoker.invoke(ActionInvoker.java:161)
>        at Invocation.HTTP Request(Play!)
> Caused by: java.io.IOException: Bad file descriptor
>        at java.io.FileInputStream.readBytes(Native Method)
>        at java.io.FileInputStream.read(FileInputStream.java:198)
>        at play.libs.IO.readContent(IO.java:180)
>        ... 8 more
> 
> ## 
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/delving/culture-hub/issues/59

--- a/app/controllers/SipCreatorEndPoint.scala
+++ b/app/controllers/SipCreatorEndPoint.scala
@@ -131,7 +131,9 @@ object SipCreatorEndPoint extends Controller with AdditionalActions {
   }
 
 
-  def acceptFile(spec: String, fileName: String): Result = {
+  def acceptFile: Result = {
+    val spec = params.get("spec")
+    val fileName = params.get("fileName")
     log.info(String.format("Accepting file %s for DataSet %s", fileName, spec))
     val dataSet = DataSet.findBySpec(spec).getOrElse(return TextError("Unknown spec %s".format(spec)))
 

